frontend/models/SignUpForm.js

```diff
@@ -0,0 +1,9 @@
+import Joi from 'joi'
+
+export const SignUpForm = Joi.object({
+  username: Joi.string().alphanum().required().max(20),
+  email: Joi.string()
+    .email({ tlds: { allow: false } })
+    .required(),
+  password: Joi.string().pattern(new RegExp('^[a-zA-Z0-9]')).min(6).required(),
+})
```

Comment:
So, are we sure this validation is all exactly as we set it up in Gitea?

Reply:
Not sure, if you added additional constraints to the password as mentioned in #21, this will need to be updated. Also don't know the max limit for username. With that said that if it passed this validation Gitea will accept it.

Reply:
Is there anyway we can unify this? Maybe not for this PR but in the future? Would be good to just define our rules in one place...

Reply:
Yes. Though I have thought of it, I think it's an antipattern; usually, models are stand-alone modules except for composite models. Also, no one validates the password for login forms. I had the same feeling though I didn't manage to come with a strong argument to make it a legit decision. Assuming the worst-case scenario the API will respond with the right error message.
I think the consistency should be considered on the testing level, otherwise, it's like writing React with Angular flavor!

Reply:
I am not sure I understand what you are proposing. What do you suggest we do?